### PR TITLE
290 - LUX is no longer dependent on generating data constants after d…

### PR DESCRIPTION
**Pull request by clarkepeterf**
…ata is loaded into the content database.

### CODE REVIEW

File: src/main/ml-modules/root/config/similarConfig.mjs [hunk 1]
```diff
@@ -87,13 +83,9 @@ const SIMILAR_CONFIG = {
...
           // this was the field agentDescriptionNoteText, but that field was removed
-          `/json[type = ('Group', 'Person')]/referred_to_by[./classified_as/id='${getIRI(
-            'biographyStatement'
-          )}']/content`,
+          `/json[type = ('Group', 'Person')]/referred_to_by[./classified_as/equivalent/id='${IDENTIFIERS.biographyStatement}']/content`,
           // there is no field configuration for this one
-          `/json[type = ('Group', 'Person')]/referred_to_by[./classified_as/id='${getIRI(
-            'descriptionStatement'
-          )}' and ./language/id='${getLanguageIRI('en')}']/content`,
+          `/json[type = ('Group', 'Person')]/referred_to_by[./classified_as/equivalent/id='${IDENTIFIERS.descriptionStatement}' and ./language/equivalent/id='${IDENTIFIERS.langen}']/content`,
```
Comment (brent-hartwig) [inline src/main/ml-modules/root/config/similarConfig.mjs:88]: @clarkepeterf, this is the only direct reference to a language identifier I found.  On purpose?

Comment (brent-hartwig) [reply src/main/ml-modules/root/config/similarConfig.mjs:88]: A choice is to export LANGUAGE_EN from model.mjs and call getLanguageIdentifier.

File: src/main/ml-modules/root/runDuringDeployment/dataConstants/dataConstantsGenerator.mjs [hunk 2]
```diff
@@ -1,349 +0,0 @@
-import { SPARQL_PREFIXES } from '../../lib/appConstants.mjs';
```
Comment (brent-hartwig) [inline src/main/ml-modules/root/runDuringDeployment/dataConstants/dataConstantsGenerator.mjs:1]: Goodbye, Data Constants Generator!  You served us well.  Best of luck on the reincarnated bytes front!

File: docs/lux-backend-repo-inventory.md [hunk 3]
```diff
@@ -24,7 +24,7 @@ What follows are brief introductory descriptions of various sub-directories with
...
+|                                |                    |                                     |                                                       | [/runDuringDeployment](/src/main/ml-modules/root/runDuringDeployment) | Includes scripts to deploy then execute during deployment, directly supporting generators for the remaining search terms, related lists, and advanced search configuration. | [LUX Gradle Tasks](/docs/lux-backend-build-tool-and-tasks.md#lux-gradle-tasks)                                                         |                                                                                                                                                                                                                           |
```
Comment (brent-hartwig) [inline docs/lux-backend-repo-inventory.md:27]: My commit will include editing the source: lux-markdown-tables.xlsx.  There was an issue with the columns too.

**Comment by brent-hartwig** [comment]
@clarkepeterf, I made it through the review but not runtime testing (Monday).  I committed my changes.  In addition to line-specific comments:

1. We should acknowledge a potential performance implication of switching from `id` to `identifier` (more triples / larger field).
2. Is it okay that nine fields still use `classified_as/id`?
3. We can remove `facetMaximumProduct` from the properties files in the CF repo.
4. At any time, we can delete the "Rejected request to calculate" [log monitoring entry](https://docs.google.com/spreadsheets/d/1uu6aL7yn047yyiZ4auujpTXnlwm01sgWZQ50ht-X4M4/edit?gid=1743835316#gid=1743835316).
5. Well done!  There was a lot to this change.  For a while I didn't think I was going to find anything to comment on.

**Comment by clarkepeterf** [comment]
Thank you @brent-hartwig 

1. True - do you think we need a performance test before releasing this?
2. I count 8 fields - is there a 9th field I'm missing? The fields I see are the <scope>TypeId fields. These are ok because they are just used for clients to search using the `id` field. Clients (including our own frontend) may still want to search using LUX URIs. As long as our code isn't dependent on a specific LUX URI we are good.
3. Good call - will do on Monday
4. Good call - will do on Monday
5. Thank you very much 😁

**Comment by clarkepeterf** [comment]
@brent-hartwig re bullet 3 above, I have removed `facetMaximumProduct` from the .properties files here:
https://git.yale.edu/lux-its/ml-cluster-formation/pull/55

**Comment by brent-hartwig** [comment]
@clarkepeterf, the changes passed all of my index generation, deployment, and runtime tests, including `{_scope:agent, similar:[iri]}` and as a unit.  More on two previous topics plus one more...

### Performance

Runtime XPath changes to equivalent/id appear limited to similarConfig.mjs and model.mjs.  Similar is yet to be tested.  I do not expect Model's XPaths to be noticeable.

That takes us to any switches from `id` to `identifier` search terms.  The former slaps the ID into a `cts.documentQuery` given the URI match.  The latter taps in to field range indexes which have between 159K (event) and 51.4M (work) values.  The most sensitive context may be semantic facets, which would use `conceptIdentifier` (5.4M values).  But we're specifying a single value.  A compounding factor is #337 having us temporarily change from `setClassifiedAs` to `referenceClassifiedAs`, until some triples can be fixed.  Some manual QC testing may be in order, but you may have already checked that box while testing #337.

Due to removing the facet maximum product threshold (discussed in #179 but implemented earlier), we may expect a performance _gain_ for non-semantic facet requests, specifically those that cts.estimate would have spent seconds on.

### classified_as/id Fields

Indeed.  Only eight.  One is configured with two XPaths.  Looks like these are only used by auto-complete and facets.

### Deployment Ports

Prior to this PR, all tasks leading up to `mlLoadModules` executed against the ML Manage port, courtesy of the `startWithAppServicesPort` task.  `mlLoadModules` needs that to insert the second app server's REST API options into the correct modules database.  The `generateDataConstants` Gradle task used to switch to port the App Services port (via `switchToAppServicesPort`) to have more time than the tenant port allowed.  That led to all of the generators running through port 8000 (and their output going in 8000_ErrorLog.txt).  They are now go through 8003 but I don't anticipate an issue, including in multi-tenant environments as the generators use the `tenant*Database` properties; e.g.:

```
xdmp.invoke(
  '/runDuringDeployment/generateAdvancedSearchConfig.mjs',
  {},
  {
    database: xdmp.database('%%tenantContentDatabase%%'),
    modules: xdmp.database('%%tenantModulesDatabase%%'),
    root: '',
  }
);
```

That said, it would be good to pay extra attention to the first couple non-main tenant deployments.

There are no remaining callers of `switchToAppServicesPort` yet I'd like to keep that task.

**Comment by clarkepeterf** [comment]
All points above noted. Merging.

**Pull request merged**